src/DependencyInjection/Configuration.php

```diff
@@ -34,7 +34,7 @@ public function getConfigTreeBuilder()
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->booleanNode('builder_alias')->defaultTrue()->end()
-                        ->booleanNode('container_aware')->defaultTrue()->end()
+                        ->booleanNode('container_aware')->setDeprecated('The "%node%" option is deprecated and will be removed in version 3.')->defaultTrue()->end()
                         ->booleanNode('builder_service')->defaultTrue()->end()
```

Comment:
that one should be deprecated for the same reason (LazyProvider also makes this service obsolete)